src/lib/operations.ts

```diff
@@ -1368,26 +1374,38 @@ export async function enterTranslateFlow({
   const selection: Selections = { graphSelections: [], otherSelections: [] }
   const argDefaultValues = { nodeToEdit, selection, x, y, z }
   return {
-    type: 'Find and select command',
-    data: {
-      name: 'Translate',
-      groupId: 'modeling',
-      argDefaultValues,
-    },
+    ...baseCommand,
+    argDefaultValues,
   }
 }
 
-export async function enterRotateFlow({
+export async function enterTranslateFlow({
   operation,
 }: EnterEditFlowProps): Promise<Error | CommandBarMachineEvent> {
+  const data = await prepareToEditTranslate({ operation })
+  if ('reason' in data) {
+    return new Error(data.reason)
```

Comment:
You don't like the `Reason` class in `trap.ts`? It's fine if you don't, but I'm just wondering why because it's exactly what you've built here.

Reply:
Ok I see, we should replace `type PrepareToEditFailurePayload = { reason: string }` with `Reason` from `trap.ts` for all prepareToEdit functions I think. But I think that should be its own PR?